packages/office-ui-fabric-react/src/components/Panel/examples/Panel.HandleDismissTarget.Example.tsx

```diff
@@ -52,15 +29,34 @@ export class PanelHandleDismissTargetExample extends React.Component<
     );
   }
 
-  private _onRenderFooterContent = (): JSX.Element => {
+  private _onRenderFooterContent = () => {
```

Comment:
Since this isn't a built-in, wonder if it would be better to correct the return type to React.ReactNode rather than removing it entirely?

Reply:
This is being mapped to a built in prop though so the return type is already implicit.  If it were a completely standalone function that was not assigned to any prop I would completely agree here.  For example if this function was simply being used in the render body alone then it would be highly beneficial to constrain the type.  However, the `_onRenderFooterContent` is already constrained, you can't return the wrong thing, so then I'd see the explicit return as redundant since it adds no value.   Two examples:

```
x = () => {
    return 1;
  }

render () {
  return x();
}
```

The above works fine, but not advisable to your point, the arrow function `x` should be typed as it can return anything right now.

However with the case of `_onRenderFooterContent` the prop it is bound to requires type:

```
export interface IRenderFunction<P> {
    (props?: P, defaultRender?: (props?: P) => JSX.Element | null): JSX.Element | null;
}
```

So if I edit that `_onRenderFooterContent` to return a 1, VSCode will show an error when editing the file that the function must return type JSX.Element, so we already have the benefits of the typing implicitly, the explicit type adds no value here.

If my above arguments don't sway you I'll add the type back, it's a relatively minor thing, let me know!

Reply:
Thinking about this more--I'm okay with omitting the return type here since it's obvious from the method name approximately what it will return (something renderable). However, if there are cases where that's *not* obvious, I think it's best for documentation purposes to specify the return type even if it would already be constrained at compile time (by either where it's passed as a prop, or a parent class definition). And if the return type is narrower than the one defined by the prop/parent, honestly that's not the end of the world and can be corrected later if needed.

Reply:
Sounds good, I'll echo the sentiments to being hugely in favor of strictly typing methods in general when they have no existing typed function they are being bound to.